src/Azure.DataApiBuilder.Mcp/BuiltInTools/ReadRecordsTool.cs

```diff
@@ -324,100 +331,5 @@ private static bool TryResolveAuthorizedRole(
             error = $"You do not have permission to read records for entity '{entityName}'.";
             return false;
         }
-
-        /// <summary>
-        /// Returns a result from the query in the case that it was successfully ran.
-        /// </summary>
-        /// <param name="entityName">Name of the entity used in the request.</param>
-        /// <param name="engineRootElement">Query result from engine.</param>
-        /// <param name="logger">MCP logger that returns all logged events.</param>
-        private static CallToolResult BuildSuccessResult(
-            string entityName,
-            JsonElement engineRootElement,
-            ILogger? logger)
-        {
-            // Build normalized response
-            Dictionary<string, object?> normalized = new()
-            {
-                ["status"] = "success",
-                ["result"] = engineRootElement // only requested values
-            };
-
-            string output = JsonSerializer.Serialize(normalized, new JsonSerializerOptions { WriteIndented = true });
-
-            logger?.LogInformation("ReadRecordsTool success for entity {Entity}.", entityName);
-
-            return new CallToolResult
-            {
-                Content = new List<ContentBlock>
-                {
-                    new TextContentBlock { Type = "text", Text = output }
-                }
-            };
-        }
-
-        /// <summary>
-        /// Returns an error if the query failed to run at any point.
-        /// </summary>
-        /// <param name="errorType">Type of error that is encountered.</param>
-        /// <param name="message">Error message given to the user.</param>
-        /// <param name="logger">MCP logger that returns all logged events.</param>
-        private static CallToolResult BuildErrorResult(
-            string errorType,
-            string message,
-            ILogger? logger)
-        {
-            Dictionary<string, object?> errorObj = new()
-            {
-                ["status"] = "error",
-                ["error"] = new Dictionary<string, object?>
-                {
-                    ["type"] = errorType,
-                    ["message"] = message
-                }
-            };
-
-            string output = JsonSerializer.Serialize(errorObj);
-
-            logger?.LogError("ReadRecordsTool error {ErrorType}: {Message}", errorType, message);
```

Comment:
By using the McpResponseBuilder's BuildErrorResult function, we are loosing the capability to log which tool errored out. Consider sending in the name of the tool to the BuildErrorResult function so we can log better.